backend/internal/service/stocks.go

```diff
@@ -46,8 +50,19 @@ func (s serviceImpl) PlaceStockOrder(userName string, stockID string, isBuy bool
 		return errors.New("invalid quantity")
 	}
 
-	if price <= 0 {
-		return errors.New("invalid price")
+	if price < 0 {
+		return errors.New("negative price")
+	} else if orderType == "MARKET" && price != 0 {
+		return errors.New("market order with non-zero price")
+	}
```

Comment:
careful here. this change is allowing LIMIT orders with price=0 which doesn't seem right

Reply:
Lol yeah; I thought about that. Decided a $0 price makes enough sense, that it's fine until proven otherwise.